server/monitor.go

```diff
@@ -789,8 +793,10 @@ type SubDetail struct {
 	Cid     uint64 `json:"cid"`
 }
 
+// Subscription client should be locked and guaranteed to be present.
 func newSubDetail(sub *subscription) SubDetail {
 	return SubDetail{
```

Comment:
ClientSubDetail is for connz and we already have CID and account, that is why I chose not to duplicate them in that one.

Reply:
I can check for the account not nil to be safe yes.

Reply:
Actually GetName() handles nil receiver.

Reply:
No, newClientSubDetail is now the one invoked by newSubsDetailList() which is invoked in Connz but also in Routez (https://github.com/nats-io/nats-server/pull/1318). Removing the field is not a good idea if users have app checking it, it will break their app. This has been available since 2.1.6

Reply:
ok fair enough.

Reply:
fixed.